syntax = "proto3";

message CreateLocationRequest {
    int32 person_id = 1;
    string longitude = 2;
    string latitude = 3;
    string creation_time = 4;
}

message CreateLocationResponse {
    string message = 1;
}

message GetLocationRequest {
    int32 location_id = 1;
}

message Location {
    int32 location_id = 1;
    int32 person_id = 2;
    string longitude = 3;
    string latitude = 4;
    string creation_time = 5;
}

message ListExposedLocationsRequest {
    int32 person_id = 1;
    string start_date = 2;
    string end_date = 3;
    int32 meters = 4;
}

message ExposedLocation {
    int32 location_id = 1;
    int32 exposed_person_id = 2;
    string exposed_long = 3;
    string exposed_lat = 4;
    string exposed_time = 5;
}

message ListExposedLocationsResponse {
    repeated ExposedLocation locations = 1;
}

service LocationService {
    rpc CreateLocation(CreateLocationRequest) returns (CreateLocationResponse);
    rpc GetLocation(GetLocationRequest) returns (Location);
    rpc ListExposedLocations(ListExposedLocationsRequest) returns (ListExposedLocationsResponse);
}
